chained/dialer.go

```diff
@@ -188,16 +193,6 @@ func defaultDialOrigin(op *ops.Op, ctx context.Context, p *proxy, network, addr
 		return nil, err
 	}
 
-	var timeout time.Duration
-	if deadline, set := ctx.Deadline(); set {
-		conn.SetDeadline(deadline)
-		// Set timeout based on our given deadline, minus a 2 second fudge factor
-		timeout := time.Until(deadline) - 2*time.Second
-		if timeout < 0 {
-			log.Errorf("Not enough time left for server to dial upstream within %v, return errUpstream immediately", timeout)
-			return nil, errUpstream
-		}
-	}
```

Comment:
This was just kind of wrong. If the context's deadline is exceeded, it's just done, so we should return. Otherwise, though, we don't want to mess around with the connection's read and write deadlines, which are separate from the context deadline, as the context deadline in net.DialContext land only governs the dial, not reads and writes, and presumably we want to adhere to that standard notion.